src/main/java/org/jabref/gui/keyboard/EditorKeyBindings.java

```diff
@@ -0,0 +1,100 @@
+package org.jabref.gui.keyboard;
+
+import java.util.Optional;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.input.KeyEvent;
+
+import org.jabref.gui.Globals;
+import org.jabref.logic.util.strings.StringManipulator;
+import org.jabref.model.util.ResultingStringState;
+import org.jabref.preferences.PreferencesService;
+
+public class EditorKeyBindings {
+
+    public static void call(Scene scene, KeyEvent event, PreferencesService preferencesService) {
+        if (scene.focusOwnerProperty().get() instanceof TextInputControl) {
+
+            KeyBindingRepository keyBindingRepository = Globals.getKeyPrefs();
+            TextInputControl focusedTextField = (TextInputControl) scene.focusOwnerProperty().get();
+            Optional<KeyBinding> keyBinding = keyBindingRepository.mapToKeyBinding(event);
+            keyBinding.ifPresent(binding -> {
+                if (binding.equals(KeyBinding.EDITOR_DELETE)) {
```

Comment:
In all other cases you use ==.
Why not refactor to a switch case?

Reply:
I did not yet work on that, had some trouble refactoring and fixing the logic part of this PR which had some defects. But thanks for the note. Next i going to take a look into getting this to work with CodeAreaa and then include a button to fill in presets (like bash keybindings).

Reply:
Good luck, for codeArea you might want to check out this_ https://github.com/FXMisc/RichTextFX/wiki/How-to-Override-the-Default-Behavior